### PR TITLE
feat(auth): support  login as impersonate user

### PR DESCRIPTION
**Pull request by xxs-wallace**
Allow impersonate as owner user. 
Developer can set email/password with environment variable TABBY_OWNER_IMPERSONATE_OVERRIDE

### CODE REVIEW

File: ee/tabby-webserver/src/service/auth.rs [hunk 1]
```diff
@@ -38,6 +38,7 @@ struct AuthenticationServiceImpl {
     mail: Arc<dyn EmailService>,
     license: Arc<dyn LicenseService>,
     setting: Arc<dyn SettingService>,
+    impersonate_user: Option<UserDAO>,
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/auth.rs:41]:
```suggestion
    impersonate_root_user_credentials: Option<ImpersonateUserCredential>,
```

Comment (xxs-wallace) [reply ee/tabby-webserver/src/service/auth.rs]: OK

File: ee/tabby-webserver/src/service/auth.rs [hunk 2]
```diff
@@ -47,27 +54,28 @@ pub fn create(
...
+    mail: Arc<dyn EmailService>,
+    license: Arc<dyn LicenseService>,
+    setting: Arc<dyn SettingService>,
+) -> AuthenticationServiceImpl {
+    let mut impersonate_user = None;
     if let Ok(value) = std::env::var("TABBY_OWNER_IMPERSONATE_OVERRIDE") {
         let words: Vec<&str> = value.split(':').collect();
         if words.len() == 2 {
-            impersonate = Some((words[0].to_string(), words[1].to_string()));
+            let password_encrypted = password_hash(words[1]).expect("can not encrypt passworkd");
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/auth.rs:70]: please fix typo

File: ee/tabby-webserver/src/service/auth.rs [hunk 3]
```diff
@@ -47,27 +54,28 @@ pub fn create(
...
+            impersonate_user = Some(ImpersonateUserCredential {
+                // impersonate user is owner use, so use 1 as id.
```
Comment (wsxiaoys) [inline ee/tabby-webserver/src/service/auth.rs:72]: Please pay more attention to english writing, e.g try optimize it with answer engine:

https://jam.dev/c/57400152-1b1a-4d6f-bb02-036c45dae34a

**Pull request merged**